### PR TITLE
Prep before SSO login flow work

### PR DESCRIPTION
**Pull request by gsoltis**
* Starts using `cobra` for the `login` command. This can serve as a model for an eventual migration to `cobra` for everything
 * `turbo login` should now respect `Ctrl+C`
 * Make golint (mostly) happy for `login.go`
 * Pass dependencies into `run` function to make it a bit more testable
 * Add a test for the happy path of logging in
 * Provides a default `cli.ColoredUi` util
 * `OpenBrowser` now returns an error rather than fatally exiting.

### CODE REVIEW

**Comment by gsoltis** [comment]
@Xenfo this PR includes an example of what I mean by gradually adopting `cobra`. It will still take some effort to wire it up to a root command in the future, but we can do that once all of the subcommands have been transitioned.

**Comment by Xenfo** [comment]
@gsoltis should I continue how I am doing it right now and figure out how to appropriately commit to `main` once the PR is ready?

**Comment by gsoltis** [comment]
@Xenfo As a reviewer, I would find it much easier to go one command per PR. Then we could do a final PR to wire them all up to a root command. 

FWIW, I'm excited to see this happening and think it's a good step. There's no reason `turbo` needs to do its own special flag parsing.

File: cli/internal/login/login.go [hunk 1]
```diff
@@ -38,67 +43,118 @@ Usage: turbo login
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(c.Config, loginDeps{
+				ui:          c.UI,
+				openURL:     browser.OpenBrowser,
+				client:      c.Config.ApiClient,
+				writeConfig: config.WriteUserConfigFile,
+			})
+		},
+	}
+	loginCommand.SetArgs(args)
```
Comment (jaredpalmer) [inline cli/internal/login/login.go:63]: :smart_meme:

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**